contracts/core/extensions/CoreInternal.sol

```diff
@@ -50,44 +50,28 @@ contract CoreInternal is
     /* ============ External Functions ============ */
 
     /**
-     * Add a factory to the mapping of tracked factories. Can only be set by
-     * owner of Core.
-     *
-     * @param  _factory   The address of the SetTokenFactory to enable
-     */
-    function enableFactory(
-        address _factory
-    )
-        external
-        onlyOwner
-    {
-        // Mark as true in validFactories mapping
-        state.validFactories[_factory] = true;
-
-        // Add to factories array
-        state.factories.push(_factory);
-    }
-
-    /**
-     * Disable a factory in the mapping of tracked factories. Can only be disabled
-     * by owner of Core.
+     * Add or remove a factory to the mapping of tracked factories. Can only be set by
+     * owner of Core
      *
-     * @param  _factory   The address of the SetTokenFactory to disable
+     * @param  _factory   Address of the contract conforming to ISetFactory
+     * @param  _enabled   Enable or disable the factory
      */
-    function disableFactory(
-        address _factory
+    function registerFactory(
+        address _factory,
+        bool _enabled
```

Comment:
This is problematic potentially, because you can call register multiple times and thus add multiple instances of the same address/factory in the array.

I think the bool `enabled` is not needed as it is implicit. If it exists, then you disable it. If not, then you add it.

Reply:
The first one is definitely a legitimate issue pre-this PR as well. +1 to that. I'm actually not sure why we keep this array since we only need the mapping for verification. It's inconsistent with our store for exchange wrappers and we've gone on fine without that list.

I'm not a fan of inversing the existing state. The clarity will permeate through to tests where it would be unclear what the current state is after an arbitrary number of `registerFactory` calls, and when we are looking back at transaction hashes.

Reply:
The array makes sense to get a list of all the available price libraries on-chain. otherwise, the only way to check is to look @ our documentation. That said, I'm not against removing the array.

Reply:
Also, there is more security for the end users if they can find a list of all enabled factories/libraries. (They would just need to trust us otherwise).
This is due to the impossibility of exploring the entire space to find which addresses are true/false.

Reply:
More security, how? I can't imagine a flow where a call to `factories`. I think when I lobbied for individual `set` arrays on the factories, we agreed that that's what logs are for. That same thinking should apply here

Reply:
I think it would be a security risk for users if they can't validate/verify what factories/libraries are active or enabled. 

Event logs could potentially be a way to check that, but we don't have that implemented. Array list works well too.

Reply:
how does this add multiple factories to an array? isnt this a mapping of factoryAddress => boolean?

Reply:
@justinkchen It doesn't anymore with the latest commit. We previously had a factories array